src/components/Account-panel/Delivery-address/Delivery-address.tsx

```diff
@@ -1,3 +1,233 @@
+import { useEffect, useState } from 'react'
+import { Box, Button, Grid, Typography } from '@mui/material'
+import CreateIcon from '@mui/icons-material/Create'
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline'
+import { Item, stBtnEdit } from '../../AddressForm/style'
+import {
+  stContainerAddress,
+  stDeleteIcon,
+  stIconM,
+  stItem,
+  stTypographyBody1,
+  stTypographyBody1Address,
+  stTypographyBody2Address
+} from '../style'
+import { stH3, stP } from '../../auth/style'
+import IconNovaPoshta from '../../../icons/novaPoshta.svg?react'
+import IconUkrPoshta from '../../../icons/ukrPoshta.svg?react'
+import axiosInstance from '../../../axios/settings'
+import fetchDataMyPostOffices from './fatchDataPostOffices'
+import {
+  CustomSnackbar,
+  ISnackbarData
+} from '../../SnackebarCustom/SnackbarCustom'
+import { ModalCustomBtnAddAddress } from '../../Modal-custom-btn/ModalCustomBtnAddAddress'
+
+type AddressInfo = {
+  icon: JSX.Element | undefined
+  text: string | undefined
+  addressHeadline: string | undefined
+  location?: string
+  titleAddress?: string
+  address_warehouse: string | null | undefined
+}
+
+export type AddressItem = {
+  id: number
+  address_warehouse: string
+  serviceType: string
+  type: string
+  addressType: string
+  source: string
+  city: string
+  street: string
+  post_code: string
+  house_number: string
+  region: string
+  apartment_number: string
+}
+
+type AddressArray = AddressItem[]
+
+const getAddressTextAndIcon = (event: AddressItem): AddressInfo => {
+  const {
+    type,
+    addressType,
+    source,
+    address_warehouse: addressWarehouse = undefined
+  } = event
+  let icon
+  let addressHeadline
+  let location
+  let titleAddress
+
+  const setIconAndHeadline = (source: string) => {
+    switch (source) {
+      case 'nova_poshta':
+        icon = <IconNovaPoshta style={stIconM} />
+        addressHeadline = 'Нова пошта'
+        break
+      case 'ukr_poshta':
+        icon = <IconUkrPoshta style={stIconM} />
+        addressHeadline = 'Укрпошта'
+        location = 'Відділення'
+        titleAddress = `${event.city}, ${event.street}, ${event.region}, ${event.post_code}, буд.${event.house_number}, кв.${event.apartment_number}`
+        break
+      default:
+        break
+    }
+  }
+
+  const setLocationAndAddress = () => {
+    if (addressType === 'адреса') {
+      location = 'Адреса'
+      titleAddress = `${event.city}, ${event.street}, буд.${event.house_number}, кв.${event.apartment_number}`
+    } else if (addressType === 'відділення') {
+      location = determineLocation(addressWarehouse ?? undefined)
+      titleAddress = `${location} ${event.address_warehouse}, ${event.city}`
+    }
+  }
+
+  setIconAndHeadline(source)
+  setLocationAndAddress()
+
+  return {
+    icon,
+    text: type,
+    addressHeadline,
+    location,
+    titleAddress,
+    address_warehouse: addressWarehouse
+  }
+}
```

Comment:
try to do cleaner logic cause now it looks like a mess. try not use let, don't repeat logic and make it more understandable

Reply:
done